### PR TITLE
Add Pages domains support

### LINKED ISSUE #966: Missing Add Custom Domain to CF Pages
### Current cloudflare-go version

latest

### Description

I cant find the Add a custom Domain for CF Pages.
It seems also missing in the API Reference, but the Endpoint seems to work: 
https://community.cloudflare.com/t/add-pages-domain-via-api/303510/4

Updating and Deleting would be awesome too

### Use cases

It would be really awesome to add custom domains. 

### Potential cloudflare-go usage

```go
api.CreatePagesDomain(...)
```

### References

_No response_

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
## Description

Add API calls for pages domains

Would resolve #966 

## Has your change been tested?

Unit tests have been added for all new functions

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.

### CODE REVIEW

File: pages_domain.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+type PagesDomainResponse struct {
+	Response
+	Result PagesDomain `json:"result,omitempty"`
+}
+
+// GetPagesDomains gets all domains for a pages project.
+//
+// API Reference: https://api.cloudflare.com/#pages-domains-get-domains
+func (api *API) GetPagesDomains(ctx context.Context, params PagesDomainsParameters) ([]PagesDomain, error) {
+	// Make sure account id is set
```
Comment (jacobbednarz) [inline pages_domain.go:66]: similar to your other PR, let's remove this style of comment from the PRs.

File: pages_domain.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	if err := json.Unmarshal(res, &pagesDomainResponse); err != nil {
+		return PagesDomain{}, err
+	}
+	return pagesDomainResponse.Result, nil
+}
+
+// PagesAddDomain adds a domain to a pages project.
+//
+// API Reference: https://community.cloudflare.com/t/add-pages-domain-via-api/303510/3
+// Blame Washly for not making it official.
```
Comment (jacobbednarz) [inline pages_domain.go:154]: @WalshyDev any chance of this documentation landing soon? i'd rather we didn't use a forum post as the API reference here.

Comment (WalshyDev) [reply pages_domain.go:154]: Yeah I can PR it in today.

Comment (jacobbednarz) [reply pages_domain.go:148]: legend! thank you!

File: pages_domain.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
```
Comment (jacobbednarz) [inline pages_domain.go:148]:
```suggestion
// API Reference: https://api.cloudflare.com/#pages-domains-add-domain
```

File: pages_domain.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	if err != nil {
+		return PagesDomain{}, err
+	}
+
+	var pagesDomainResponse PagesDomainResponse
+	if err := json.Unmarshal(res, &pagesDomainResponse); err != nil {
+		return PagesDomain{}, err
+	}
+	return pagesDomainResponse.Result, nil
+}
```
Comment (jacobbednarz) [inline pages_domain.go:173]: let's add `Delete` here too; the fragment will be `#pages-domains-delete-domain`

```
DELETE accounts/:account_id/pages/projects/:project_name/domains/:domain_name
```

File: pages_domain.go (hunk 2 above)
Comment (jacobbednarz) [reply pages_domain.go:148]: i'm tracking this internally so we're 👍 to merge this when we address the couple of feedback items.

File: pages_domain_test.go [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+	"time"
+)
+
+const testPagesProjectName = "page-project"
+
+var testCreatedOn, _ = time.Parse(time.RFC3339, "2017-01-01T00:00:00Z")
+
+var testPagesDomain = PagesDomain{
+	ID:     "8232210c-6818-4e34-8d95-cc386874b8d2",
+	Name:   "washly.mvp",
```
Comment (jacobbednarz) [inline pages_domain_test.go:18]: let's use documentation domains (ending in `.test`) or `example.com` instead of real domains.

**Review by jacobbednarz — APPROVED** [review]

**Comment by jacobbednarz** [comment]
thanks again for another awesome PR!

**Pull request merged**